### PR TITLE
fix: hidden ldf namespace

### LINKED ISSUE #3708: action namespace started with "l" is breaking everything
#### Describe the bug
Predicting is broken when action have namespace started with 'l'

#### To Reproduce
cb-a.txt
```
{"_label_cost":-1,"_label_probability":0.5,"_label_Action":2,"_labelIndex":1,"a":[2,1],"c":{"shared":{"f":"1","_multi":[{"a":{"f":"0"}},{"a":{"f":"1"}}]}},"p":[0.5,0.5]}
```
cb-l.txt
```
{"_label_cost":-1,"_label_probability":0.5,"_label_Action":2,"_labelIndex":1,"a":[2,1],"c":{"f":"1","_multi":[{"l*":{"f":"0"}},{"l*":{"f":"1"}}]},"p":[0.5,0.5]}
```
The only difference between these examples is action namespace ('a' vs 'l').
1) Run:
```
vw --cb_explore_adf --dsjson -d cb-a.txt -p pred-a.txt
```
stdout:
```
predictions = pred-a.txt
using no cache
Reading datafile = cb-a.txt
num sources = 1
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
Enabled reductions: gd, scorer-identity, csoaa_ldf-rank, cb_adf, cb_explore_adf_greedy, shared_feature_merger
Input label = cb
Output pred = action_probs
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
-1.000000 -1.000000            1            1.0 1:-1:0.5        0:0.5...        6

finished run
number of examples = 1
weighted example sum = 1.000000
weighted label sum = 0.000000
average loss = -1.000000
total feature number = 6
```
pred-a.txt:
```
0:0.5,1:0.5
```

2) Run:
```
vw --cb_explore_adf --dsjson -d cb-l.txt -p pred-l.txt
```
stdout:
```
predictions = pred-l.txt
using no cache
Reading datafile = cb-l.txt
num sources = 1
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
Enabled reductions: gd, scorer-identity, csoaa_ldf-rank, cb_adf, cb_explore_adf_greedy, shared_feature_merger
Input label = cb
Output pred = action_probs
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0 1:-1:0.5 no action        6

finished run
number of examples = 1
weighted example sum = 1.000000
weighted label sum = 0.000000
average loss = 0.000000
total feature number = 6
```
pred-l.txt:
```
```

#### Expected behavior
Output for cb-l.txt should be similar to cb-a.txt and predictions output should not be empty

#### Observed Behavior
Predictions are corrupted (see 'no action' in stdout and empty predictions file)

#### Environment
vw 9.0 on win11

#### Additional context
Looks like the problem is with this guy:
https://github.com/VowpalWabbit/vowpal_wabbit/blob/41dbe3e316065c20ee9d42e71f7f0a9aa55f82c5/vowpalwabbit/csoaa_ldf.cc#L51
If we have namespace started with 'l' and it is in indices[0], we have false positive output from this function which leads to exclusion of first example from the rest of prediction stack. So, for n actions we just output n-1 predictions which later leads to some undefined issues later (I was able to observe silent wrong behavior like described before for cb and abort() for the same issue in ccb)

### PR DESCRIPTION
**Pull request by ataymano**
Fixes #3708

### CODE REVIEW

**Comment by ataymano** [comment]
Fixes #3708

File: vowpalwabbit/constant.h [hunk 1]
```diff
@@ -34,6 +34,7 @@ constexpr unsigned char node_id_namespace = 136;                   // this is \x
 constexpr unsigned char baseline_enabled_message_namespace = 137;  // this is \x89
 constexpr unsigned char ccb_slot_namespace = 139;
 constexpr unsigned char ccb_id_namespace = 140;
+constexpr unsigned char ldf_namespace = 141;
```
Comment (lalo) [inline vowpalwabbit/constant.h:37]: I'm guessing this should be excluded from AutoML too?, https://github.com/VowpalWabbit/vowpal_wabbit/blob/912fbd2765f67aea2675352b5aefd42dcc4b5f52/vowpalwabbit/reductions/automl.h#L41

Comment (lalo) [reply vowpalwabbit/constant.h:37]: Actually this might be below cb_adf so there might not be a need.

**Comment by ataymano** [comment]
this code can be completely removed - PR will be recreated

**Pull request closed without merging**